content/en/tracing/trace_collection/custom_instrumentation/android/otel.md

```diff
@@ -0,0 +1,482 @@
+            }
+
+            override fun getPropagators(): ContextPropagators {
+                return ContextPropagators.noop()
+            }
+        })
+ // and later on if you want to access the tracer provider
+ val tracerProvider = GlobalOpenTelemetry.get().getTracer(instrumentationName = "<instrumentation_name>")
+```
+{{% /tab %}} 
+{{% tab "Java" %}}
+```java
+GlobalOpenTelemetry.set(new OpenTelemetry() {
+            private final TracerProvider tracerProvider = new OtelTracerProvider.Builder()
+                    .setService(BuildConfig.APPLICATION_ID)
+                    .build();
+
+            @Override
+            public TracerProvider getTracerProvider() {
+                return tracerProvider;
+            }
+
+            @Override
+            public ContextPropagators getPropagators() {
+                return ContextPropagators.noop();
+            }
+        };
+ // and later on if you want to access the tracer provider
+ final TracerProvider tracerProvider = GlobalOpenTelemetry.get().getTracer("<instrumentation_name>");       
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+Note: The `setService` method is used to set the service name for the tracer provider. The service name is used to identify the application in the Datadog UI. You can either use the GlobalOpenTelemetry to hold a single instance of the TracerProvider create your own instance of the TracerProvider and use it in your application code as needed.
+
+4. Instrument your code with the OpenTelemetry API:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+val span = tracer.spanBuilder(spanName = "<span_name>").startSpan()
+// do something you want to measure ...
+// ... then, when the operation is finished:
+span.end()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+final Span span = tracer.spanBuilder("<span_name>").startSpan();
+// do something you want to measure ...
+// ... then, when the operation is finished:
+span.end();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+5. (Optional) Set child-parent relationship between your spans:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+let childSpan = tracer.spanBuilder(spanName = "response decoding")
+    .setParent(Context.current().with(parentSpan)) // make it child of parent span
+    .startSpan()
+
+// ... do your logic here ...
+childSpan.end()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+final Span childSpan = tracer.spanBuilder("<span_name>")
+    .setParent(Context.current().with(parentSpan)) // make it child of parent span
+    .startSpan();
+
+// ... do your logic here ...
+childSpan.end();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+6. (Optional) Provide additional attributes alongside your span:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+tracer.spanBuilder(spanName = "<span_name>").setAttribute(key = "<key_name>", value = <key_value>).startSpan()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+tracer.spanBuilder("<span_name>").setAttribute("<key_name>", <key_value>).startSpan();
```

Comment:
```suggestion
tracer.spanBuilder("<span_name>").setAttribute("<key_name>", "<key_value>").startSpan();
```

Reply:
nope because if we put it like that will mean that <key_value> should always be a string which is not the case at least in Android API